### ISSUE TITLE
Panning behaving differently if I upgrade from v3.1.0 to v4.1.0

### ISSUE THREAD

**Issue by AnujDivkar** [openseadragon/openseadragon#2376]
I have updated the openseadragon version from v3.1.0 to v4.1.0 in my react app.
Suddenly the initial panning is behaving differently.
I want the panning to enable only on zooming as in V3.1.0.
Which was possible with setting
constrainDuringPan: true,
visibilityRatio: 1.0

Now the same is not working in v4.1.0.

If u try to pan the image vertically it behaves differently.
Changed the version to 4.1.0 in package.json

tried preserveImageSizeOnResize : true

```
import OpenSeadragon from "openseadragon"
function App() {
  let duomo = {
    Image: {
      xmlns: "http://schemas.microsoft.com/deepzoom/2008",
      Url: "https://openseadragon.github.io/example-images/duomo/duomo_files/",
      Format: "jpg",
      Overlap: "2",
      TileSize: "256",
      Size: {
        Width: "13920",
        Height: "5000"
      }
    }
  };

  let viewer = OpenSeadragon({
    id: "seadragon-viewer",
    prefixUrl: "https://openseadragon.github.io/openseadragon/images/",
    tileSources: duomo,
    constrainDuringPan: true,
    visibilityRatio: 1.0,
    preserveImageSizeOnResize: true
  });
}
export default App;
```

Demo is available at Stackblitz urls:
[v3.1.0](https://stackblitz.com/edit/osd-3-1-0?file=package.json)
[v4.1.0](https://stackblitz.com/edit/osd-4-1-0?file=package.json)

**Comment by AnujDivkar** [comment]
Note:  preserveImageSizeOnResize: true works in basic html using cdn script. The issue occurs only in react application

**Comment by pearcetm** [comment]
@AnujDivkar can you put together a codepen demo (or similar) that shows the new behavior? I don’t personally use react so you might have to dig into that aspect of it further to help figure this out.

**Comment by AnujDivkar** [comment]
@pearcetm The Demo is available at below Stackblitz urls:
[v3.1.0](https://stackblitz.com/edit/osd-3-1-0?file=package.json)
[v4.1.0](https://stackblitz.com/edit/osd-4-1-0?file=package.json)

**Comment by pearcetm** [comment]
The difference in panning that you're noticing is related to https://github.com/openseadragon/openseadragon/pull/2160, which fixed incorrect boundary constraint logic that kept the image centered when it wasn't filling the viewport. To reproduce it in 4.1.0 you might need to do something like add an event listener to the `pan` and/or `viewport-change` events and do the calculations to keep it centered.

**Comment by AnujDivkar** [comment]
@pearcetm I am able to center the image using
```
viewer.addHandler('viewport-change', () => {
    viewer.viewport.fitBounds(viewer.viewport.getHomeBounds(),true);
});
```
But I cannot pan and zoom now keeping the image cantered.
Could you kindly guide me on going ahead.

**Comment by pearcetm** [comment]
Yes, the code you have would just lock the image at the home bounds. You'll need something a bit more complex to set the `y` position only and let the rest of pan/zoom continue:

```
viewer.addHandler('viewport-change', () => {
    let viewportBounds = viewer.viewport.getBounds();
    let imageBounds = viewer.world.getItemAt(0).getBounds(true);

    if (viewportBounds.height > imageBounds.height) {
      let target = viewer.viewport.getCenter(false);
      let current = viewer.viewport.getCenter(true);
      let immediate =
        target.x === current.x &&
        target.y === current.y &&
        viewer.viewport.getZoom(true) === viewer.viewport.getZoom(false);
      let yOffset = viewportBounds.getCenter().y - imageBounds.getCenter().y;
      viewportBounds.y -= yOffset;
      viewer.viewport.panTo(
        new OpenSeadragon.Point(
          current.x,
          viewportBounds.y + viewportBounds.height / 2
        ),
        true
      );
      viewer.viewport.fitBounds(viewportBounds, immediate);
    }
  });
```

**Comment by btzr-io** [comment]
Is there anyway to recreate the old behavior on v6.0.1 ? I try to overwrite the pan position on  viewport-change but it doesn't feel smooth or responsive as in v.3.0.1 as the image moves a lot or jumps position, I guess I need to overwrite the zoom out function instead somehow so it always zooms out to the center or is there an option already to force this behavior ?

**Comment by iangilman** [comment]
@btzr-io These options seem to be working for me: 

```
constrainDuringPan: true,
visibilityRatio: 1.0,
```

How about you?

**Comment by btzr-io** [comment]
This is not longer working as expected after v3.1.0, I need to keep the image on center when it zooms out and don't allow pan if is fully visible ( withing viewports bounds ).

```
constrainDuringPan: true,
visibilityRatio: 1.0,
```

Here is an example of the old behavior in 3.1.0:
- After zoom out the image returns to the center.
-  You can't pan the image if is fully visible.

https://codepen.io/btzr-io/pen/XJjNRqB

Version 6.2.0:
- After you zoom out you can see it not longer returns to center using mouse or other gestures to zoom to a specific point.
- You can still pan the image when the image is fully visible so any interaction will make the image move and never keep it on center.
https://codepen.io/btzr-io/pen/raMWOvB

**Comment by iangilman** [comment]
I see; do you want the image centered to stay centered. Give this a try:

https://codepen.io/iangilman/pen/JoPbEpX

... but add these options to it:

```
visibilityRatio: 1.0,
constrainDuringPan: true
```

You can also add:

```
minZoomImageRatio: 1
```

…so there's less white space when you're zoomed out.
